files/start.sh

```diff
@@ -24,6 +24,13 @@ if [[ ! -z "${DEBUG_MODE}" ]]; then
         syslog-ng
     fi
 fi
+
+#execute pre-init scripts if this is an s2i executed 
+if [[ -d "${APP_DATA}/src/postfix-pre-init" ]]; then
+    echo "S2I is enabled, sourcing pre-init scripts..."
+    process_extending_files ${APP_DATA}/src/postfix-pre-init
```

Comment:
I am thinking why `src/postfix-pre-init`, why not just `src/`. Is the reason is it can be extended also to `post-init` in future?

Reply:
I was trying to follow sort of an standard set by SCL so it would be less confusing for people. See, in SCL images, they have a directory structure like this:

| Folder name       | Description                |
|-------------------|----------------------------|
| `./${IMAGE_NAME}-cfg`     | Can contain additional configuration files |
| `./${IMAGE_NAME}-pre-init`| Can contain shell scripts (`*.sh`) that are sourced before `${IMAGE_NAME}` is started|
| `./`              | Application source code |

For reference see https://github.com/sclorg/httpd-container/tree/master/2.4#usage
or https://github.com/sclorg/mongodb-container/tree/master/3.2#extending-image

While I agree there is probably no app source code, I'd still like to do it this way, for easier use.

Reply:
Okay, sounds good.